src/index.ts

```diff
@@ -556,15 +562,15 @@ export function runAsWorker<
   T extends AnyAsyncFn<R> = AnyAsyncFn<R>,
 >(fn: T) {
   // type-coverage:ignore-next-line -- we can't control
-  if (!workerData) {
+  if (!workerData || !parentPort) {
```

Comment:
```suggestion
  if (!workerData) {
```

It's must run in a worker.